src/theory/arith/congruence_manager.h

```diff
@@ -155,9 +155,28 @@ class ArithCongruenceManager {
   bool propagate(TNode x);
   void explain(TNode literal, std::vector<TNode>& assumptions);
 
-
+  /** Assert this literal to the eq engine. Common functionality for
+   *   * assertionToEqualityEngine(..)
+   *   * equalsConstant(c)
+   *   * equalsConstant(lb, ub)
+   * If proofNew is off, then just asserts.
+   */
+  void assertLitToEqualityEngine(Node lit,
+                                 TNode reason,
+                                 std::shared_ptr<ProofNode> pf);
   /** This sends a shared term to the uninterpreted equality engine. */
-  void assertionToEqualityEngine(bool eq, ArithVar s, TNode reason);
+  void assertionToEqualityEngine(bool eq,
+                                 ArithVar s,
+                                 TNode reason,
+                                 std::shared_ptr<ProofNode> pf);
+
+  /** Check for proof for this or a symmetric fact
+   *
+   * @returns whether this or a symmetric fact has a proof.
+   */
+  bool hasProofFor(TNode f) const;
+  /** Sets the proof for this fact and the symmetric one. */
```

Comment:
Can you say a bit more about this method: when/who should call this? It is SAT context dependent?

Reply:
Good idea. I dropped a pointer here to the documentation for the proof generators, whose documentation describes the closure properties of the different kinds of proofs at play in `ArithCongruenceManager`.